server/api.js

```diff
@@ -168,28 +168,45 @@ client.connect(function () {
 		res.send({ status:"success" });
 	});
 
-	router.put("/students/:id/feedback/:feedbackId", function(req, res) {
-
-
-		// const collection = db.collection("StudentDenormalizedData");
+	router.put("/students/:studentId/feedback/:feedbackId", (req, res) => {
+		const collection = db.collection("StudentDenormalizedData");
+		// check if the id is valid if not -> 404
 		const data = req.body;
-		console.log({ data });
-
-		// const id = new mongodb.ObjectID(req.params.id);
-		// const queryObject = { _id: id };
-
-		// const feedbackId = req.params.feedbackId;
+		if (!mongodb.ObjectID.isValid(req.params.studentId)) {
+			return res.send(404);
+		}
+		//define id (mongodb.ObjectID)
+		const studentId = new mongodb.ObjectID(req.params.studentId);
+		const queryObject = { _id: studentId };
+		const options = { returnOriginal: false }; // send back the UPDATED record
 
-		// const options = { multi:true };
+		collection.findOne(queryObject, (error, result) => {
```

Comment:
Why are we using findOne as well as findOneAndUpdate inside the put request? 
I think findOne is usually just to GET the first document but it is not used for updating. 
Also, because the update is happening after the findOne so I assume its purpose is only to get the existing feedback for a certain student. 
In this case we could just use the query object in findOneAndUpdate to get that student and then we will have access to all feedback. 
Or maybe use amending the properties of updatedFeedback inside the findOneAndUpdate method instead.

Reply:
I am happy if you come up with another way but this was the only way made it worked. I am still on it to improve but since it is a working code, can we keep it for the sake of moving forward!

Reply:
The difficulty I came across when using findOneandUpdate or Update was; it was updating the document and removing all other documents from that allFeedback. The one above, finds the allfeedback of the given student at first and applies the requested edit to the matching feedback.id and then updating the allFeedback with FindOneAndUpdate. As I said above, I am sure there are a number of techniques to do the same job but this is the one I managed it work. Please feel free to come up with better solution. I am more than happy to learn better ways.

Reply:
It shouldn't remove all the other documents, the job of findOneAndUpdate is just that. 
There should be a way to target only the specific object that you want to update from inside the array. 
Maybe spread the containing array if needed not to change the rest.